theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer.tsx

```diff
@@ -286,12 +295,13 @@ export class TraceViewerWidget extends ReactWidget {
     protected onOutputRemoved(outputId: string): void {
         const outputToKeep = this.outputDescriptors.filter(output => output.id !== outputId);
         this.outputDescriptors = outputToKeep;
-        // This is a temporary fix to not delete categories when the output is removed
-        // and hence commenting this code for now.
 
-        /* const payload: { [key: string]: {categoryName: string, toggleInd: boolean}[] } = {};
-        payload[outputId] = [];
-        signalManager().fireAnnotationsFetchedSignal(payload); */
+        const payload = {
+            traceViewerId: this.id,
+            outputId: outputId,
+            annotationCategories: []
```

Comment:
I'm not sure if this is the cause, but when I have two outputs opened for the same trace, and I close one output, the Markers filter button disappears even though there are markers in the remaining output.